src/metabase/db/migrations.clj

```diff
@@ -149,3 +150,23 @@
       (k/set-fields {:visibility_type "normal"})
       (k/where      {:visibility_type "unset"
                      :active          true}))))
+
+
+;; deal with dashboard cards which have NULL `:row` or `:col` values
+(defmigration fix-dashboard-cards-without-positions
+  (let [bad-dashboards (k/select DashboardCard (k/fields [:dashboard_id]) (k/modifier "DISTINCT") (k/where (or (= :row nil) (= :col nil))))]
+    (when (not-empty bad-dashboards)
```

Comment:
probably better to use `(when-not (empty? ...) ...)` here, `not-empty` is more for making sure you get a non-empty collection, less for predicate use (that's why it doesn't end in a `?`)